src/t_list.c

```diff
@@ -501,7 +501,7 @@ void popGenericCommand(client *c, int where) {
             return;
     }
 
-    robj *o = lookupKeyWriteOrReply(c, c->argv[1], shared.null[c->resp]);
+    robj *o = lookupKeyWriteOrReply(c, c->argv[1], hascount ? shared.nullarray[c->resp]: shared.null[c->resp]);
```

Comment:
I don't think we should complicate this. Users should not rely on which kind of null is returned.

The [protocol documentation](https://redis.io/topics/protocol) says:

> The concept of Null Array exists as well, and is an alternative way to specify a Null value (usually the Null Bulk String is used, but for historical reasons we have two formats).

Reply:
@zuiderkwast i don't think this is a serious complication. the breaking change bothers me more (maybe the doc quote you posted reduces that).

considering the documentation we wrote for 6.2 (added in the top comment), the response should be a null array, and it seems more logical to fix the code rather than the doc.